tests/integration/test_sns.py

```diff
@@ -142,6 +142,44 @@ def test_filter_policy(self):
         num_msgs_2 = len(self.sqs_client.receive_message(QueueUrl=self.queue_url_2, VisibilityTimeout=0)['Messages'])
         self.assertEqual(num_msgs_2, num_msgs_1)
 
+    def test_data_type_sns_message_int(self):
+        queue_arn = aws_stack.sqs_queue_arn(TEST_QUEUE_NAME_2)
+        filter_policy = {'attr1': [{'numeric': ['>', 0, '<=', 100]}]}
+        self.sns_client.subscribe(
+            TopicArn=self.topic_arn,
+            Protocol='sqs',
+            Endpoint=queue_arn,
+            Attributes={
+                'FilterPolicy': json.dumps(filter_policy)
+            }
+        )
+
+        # publish message that satisfies the filter policy, assert that message is received
+        message = u'This is a test message'
+        self.sns_client.publish(TopicArn=self.topic_arn, Message=message,
+                                MessageAttributes={'attr1': {'DataType': 'Number', 'StringValue': '99'}})
+        messages = self.sqs_client.receive_message(QueueUrl=self.queue_url_2, VisibilityTimeout=0)['Messages']
+        self.assertEqual(json.loads(messages[0]['Body'])['MessageAttributes']['attr1']['Value'], 99)
+
+    def test_data_type_sns_message_float(self):
+        queue_arn = aws_stack.sqs_queue_arn(TEST_QUEUE_NAME_2)
+        filter_policy = {'attr1': [{'numeric': ['>', 0, '<=', 100]}]}
+        self.sns_client.subscribe(
+            TopicArn=self.topic_arn,
+            Protocol='sqs',
+            Endpoint=queue_arn,
+            Attributes={
+                'FilterPolicy': json.dumps(filter_policy)
+            }
+        )
+
+        # publish message that satisfies the filter policy, assert that message is received
+        message = u'This is a test message'
+        self.sns_client.publish(TopicArn=self.topic_arn, Message=message,
+                                MessageAttributes={'attr1': {'DataType': 'Number', 'StringValue': '99.12'}})
+        messages = self.sqs_client.receive_message(QueueUrl=self.queue_url_2, VisibilityTimeout=0)['Messages']
+        self.assertEqual(json.loads(messages[0]['Body'])['MessageAttributes']['attr1']['Value'], 99.12)
```

Comment:
Same as above - assertion should test for string value `'99.12'` (in fact, we may be able to remove this test, as there should be no real difference between int and float).